compiler/src/dotty/tools/dotc/typer/Applications.scala

```diff
@@ -1982,13 +1975,19 @@ trait Applications extends Compatibility {
         // alternatives are the same after following ExprTypes, pick one of them
         // (prefer the one that is not a method, but that's arbitrary).
         if alt1.widenExpr =:= alt2 then -1 else 1
-      else ownerScore match
-        case  1 => if winsType1 || !winsType2 then  1 else 0
-        case -1 => if winsType2 || !winsType1 then -1 else 0
-        case  0 =>
-          if winsType1 != winsType2 then if winsType1 then 1 else -1
-          else if alt1.symbol == alt2.symbol then comparePrefixes
-          else 0
+      else
+        // For implicit resolution, take ownerscore as more significant than type resolution
+        // Reason: People use owner hierarchies to explicitly prioritize, we should not
+        // break that by changing implicit priority of types. On the other hand we do want
+        // to comparePrefixes if there is a draw; StringFormaterTest breaks if we don't do that.
+        def drawOrOwner = if preferGeneral then ownerScore else 0
+        ownerScore match
+          case  1 => if winsType1 || !winsType2 then  1 else drawOrOwner
+          case -1 => if winsType2 || !winsType1 then -1 else drawOrOwner
+          case  0 =>
```

Comment:
```suggestion
          case  _ =>
```
This wasn't introduced by the changes here, but should `(ownerScore = 0, winsType1 = false, winsType2 = true)` not also enter that case ?
i.e. the match may not be exhaustive

Reply:
ownerScore is one of -1, 0, +1, so the change would not do anything.

Reply:
Right, for some reason I had been looking at the match as if the `if` expressions were pattern guards 😔